src/test/java/seedu/address/model/person/ContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,153 @@
+
+        // Non-matching keyword
+        ArgumentMultimap mapForNonMatchingKeyword = new ArgumentMultimap();
+        mapForNonMatchingKeyword.put(PREFIX_NAME, "Carol");
+        predicate = new ContainsKeywordsPredicate(mapForNonMatchingKeyword);
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
+
+        // Keywords match phone, email and address, but does not match name
+        List<String> notMatchingNameKeywords = Arrays.asList("12345", "alice@email.com", "Main", "Street");
+        ArgumentMultimap mapForNotMatchingNameKeywords = new ArgumentMultimap();
+        mapForNotMatchingNameKeywords.put(PREFIX_PHONE, "12345");
+        mapForNotMatchingNameKeywords.put(PREFIX_EMAIL, "alice@email.com");
+        mapForNotMatchingNameKeywords.put(PREFIX_ADDRESS, "Main");
+        mapForNotMatchingNameKeywords.put(PREFIX_ADDRESS, "Streer");
+        predicate = new ContainsKeywordsPredicate(mapForNotMatchingNameKeywords);
+        assertFalse(predicate.test(new PersonBuilder().withName("Alice").withPhone("12345")
+                .withEmail("alice@email.com").withAddress("Main Street").build()));
+    }
+
+    @Test
+    public void toStringMethod_withName() {
+        List<String> keywords = List.of("keyword1", "keyword2");
+        ArgumentMultimap mapForKeywords = new ArgumentMultimap();
+        keywords.stream().forEach(keyword -> mapForKeywords.put(PREFIX_NAME, keyword));
+        ContainsKeywordsPredicate predicate = new ContainsKeywordsPredicate(mapForKeywords);
+
+        String expected = ContainsKeywordsPredicate.class.getCanonicalName() + "{n/=" + keywords + "}";
+        assertEquals(expected, predicate.toString());
+    }
+
+    @Test
+    public void toStringMethod_withTags() {
+        List<String> names = List.of("Alice", "Bob");
+        List<String> tags = List.of("friend", "family");
+
+        ArgumentMultimap mapForKeywords = new ArgumentMultimap();
+        names.forEach(name -> mapForKeywords.put(PREFIX_NAME, name));
+        tags.forEach(tag -> mapForKeywords.put(PREFIX_TAG, tag));
+
+        ContainsKeywordsPredicate predicate = new ContainsKeywordsPredicate(mapForKeywords);
+
+        String expected = ContainsKeywordsPredicate.class.getCanonicalName()
+                + "{n/=" + names + ", t/=" + tags + "}";
+
+        assertEquals(expected, predicate.toString());
+    }
+
+    @Test
+    public void toStringMethod_withRole() {
+        List<String> roles = List.of("patient", "caregiver");
+
+        ArgumentMultimap mapForKeywords = new ArgumentMultimap();
+        roles.forEach(role -> mapForKeywords.put(PREFIX_ROLE, role));
+
+        ContainsKeywordsPredicate predicate = new ContainsKeywordsPredicate(mapForKeywords);
+
+        String expected = ContainsKeywordsPredicate.class.getCanonicalName()
+                + "{role/=" + roles + "}";
+
+        assertEquals(expected, predicate.toString());
+    }
+
+}
```

Comment:
So far everything looks good to me. You just got to cover a few more methods with test cases and you would be done.